src/chunked_scatter/safe_scatter.py

```diff
@@ -0,0 +1,244 @@
+        if first_time:
+            current_bin: List[BedRegion] = [region]
+            current_bin_size = len(region)
+            first_time = False
+            continue
+        # If adding this region would put us over the target bin size,
+        # yield the bin and start a new one
+        if current_bin_size + len(region) > target_bin_size:
+            # Here we merge the chunks back together if they are adjacent
+            yield list(merge_regions(current_bin))
+            current_bin = [region]
+            current_bin_size = len(region)
+        # If this region does not put us over the target bin size, add it
+        else:
+            current_bin.append(region)
+            current_bin_size += len(region)
+    # If we are done, yield the last bin
+    yield list(merge_regions(current_bin))
+
+
+def argument_parser() -> argparse.ArgumentParser:
+    """Argument parser for the scatter-regions program."""
+    parser = common_parser()
+    parser.description = (
+        "Given a sequence dict, fasta index or a bed file, scatter over the "
+        "defined contigs/regions. Creates a bed file where the contigs add up "
+        "to the average scatter size to within min_scatter_size. NOTE, this "
+        "tool always splits up contigs.")
+    parser.formatter_class = argparse.ArgumentDefaultsHelpFormatter
+    parser.add_argument("-c", "--scatter-count", type=int, required=True,
+                        help="The number of chunks to scatter the regions in. "
+                             "All chunks will be within --min-scatter-size "
+                             "of each other except for the final chunk.")
+    parser.add_argument("-m", "--min-scatter-size", type=int,
+                        default=10000,
+                        help="The minimum size of a scatter. This tool will "
+                             "never generate regions smaller than this "
+                             "value, unless the original regions are"
+                             "smaller.")
+    parser.add_argument("--mix-small-regions", default=False,
+                        action="store_true",
+                        help=(
+                            "Mix small regions in between regular regions. "
+                            "This can be useful in case there "
+                            "is a bias in the composition of the regions. For "
+                            "example, the human reference genome has all "
+                            "unplaced contigs (which are small and difficult "
+                            "to process) at the end of the file, which means "
+                            "they all end up in the same bedfile. Enabling "
+                            "mixing prevents this. Small regions defined as "
+                            "regions that will not be split up by the "
+                            "scattering."
+                        ))
+    return parser
+
+
+def main():
```

Comment:
You can add a simple test that tests the main function. That should improve your coverage problem while also testing the main logic (which is good to test as well).

See examples in test_main.py

Reply:
Test coverage is now 100% for safe_scatter, as it should be :monocle_face:.